services/harborregistry/harborregistry.yml

```diff
@@ -0,0 +1,255 @@
+apiVersion: v1
+kind: Template
+metadata:
+  name: harborregistry-template
+  annotations:
+    description: "A template for generating Harbor's registry service"
+parameters:
+  - name: SERVICE_NAME
+    description: Name of this service
+    required: true
+  - name: SAFE_BRANCH
+    description: Which branch this belongs to, special chars replaced with dashes
+    required: true
+  - name: SAFE_PROJECT
+    description: Which project this belongs to, special chars replaced with dashes
+    required: true
+  - name: BRANCH
+    description: Which branch this belongs to, original value
+    required: true
+  - name: PROJECT
+    description: Which project this belongs to, original value
+    required: true
+  - name: LAGOON_GIT_SHA
+    description: git hash sha of the current deployment
+    required: true
+  - name: SERVICE_ROUTER_URL
+    description: URL of the Router for this service
+    value: ""
+  - name: OPENSHIFT_PROJECT
+    description: Name of the Project that this service is in
+    required: true
+  - name: REGISTRY
+    description: Registry where Images are pushed to
+    required: true
+  - name: DEPLOYMENT_STRATEGY
+    description: Strategy of Deploymentconfig
+    value: "Rolling"
+  - name: HARBORREGISTRY_SERVICE_IMAGE
+    description: Pullable image of registry service
+    required: true
+  - name: HARBORREGISTRYCTL_SERVICE_IMAGE
+    description: Pullable image of registryctl service
+    required: true
+  - name: HARBORREGISTRY_SERVICE_NAME
+    description: Pullable image of registry service
+    value: harborregistry
+  - name: HARBORREGISTRYCTL_SERVICE_NAME
+    description: Pullable image of registryctl service
+    value: harborregistryctl
+objects:
+- apiVersion: v1
+  kind: Secret
+  metadata:
+    name: ${SERVICE_NAME} # harborregistry
+    labels:
+      service: ${SERVICE_NAME}
+      branch: ${SAFE_BRANCH}
+      project: ${SAFE_PROJECT}
+  type: Opaque
+  data:
+    REGISTRY_REDIS_PASSWORD: ""
+    #REGISTRY_STORAGE_S3_ACCESSKEY: "" # Will be injected via GraphiQL at build time
```

Comment:
I don't think the comment here is correct. `GraphiQL` cannot inject anything here in build time. Can you explain where this env variable really comes from?

Reply:
This comment was meant to say that that variable will be stored via the API, so we don't have to track it here. I wanted to make sure to leave a commented version of that variable, tho, so we'd know later that it was being consumed here.